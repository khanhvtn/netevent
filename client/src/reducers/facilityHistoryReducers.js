import {
  FACILITY_HISTORY_LOADING,
  FACILITY_HISTORY_GET_ALL,
  FACILITY_HISTORY_GET_ALL_FILTER
} from '../constants';

const initialState = {
  isLoading: false,
  facilityHistories: [],
  totalPages: null,
  createSuccess: false,
  updateSuccess: false,
  deleteSuccess: false
};

export default function facilityHistoryReducers(state = initialState, action) {
  switch (action.type) {
    case FACILITY_HISTORY_GET_ALL:
      return {
        ...state,
        facilityHistories: action.payload.data.data
      };
    case FACILITY_HISTORY_GET_ALL_FILTER:
      return {
        ...state,
        facilityHistories: action.payload.data.data,
        totalPages: action.payload.data.totalPages
      };
    case FACILITY_HISTORY_LOADING:
      return { ...state, isLoading: action.payload };
    default:
      return state;
  }
}
