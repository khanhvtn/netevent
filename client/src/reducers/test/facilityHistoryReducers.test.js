import {
    FACILITY_HISTORY_LOADING,
    FACILITY_HISTORY_GET_ALL,
    FACILITY_HISTORY_GET_ALL_FILTER,
} from '../../constants';
import facilityHistoryReducers from '../facilityHistoryReducers'
import expect from 'expect'

describe('FACILITY HISTORY REDUCER', () => {
    const initialState = {
        isLoading: false,
        facilityHistories: [],
        totalPages: null,
        createSuccess: false,
        updateSuccess: false,
        deleteSuccess: false,
    };
    it('Should return default state', () => {
        const action = facilityHistoryReducers(undefined, {})
        expect(action).toMatchSnapshot(initialState)
    });
    it('Should return FACILITY_HISTORY_LOADING', () => {
        const action = facilityHistoryReducers(initialState, {
            type: FACILITY_HISTORY_LOADING
        })
        expect(action).toMatchSnapshot({
            isLoading: action.payload,
            facilityHistories: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: false,
            deleteSuccess: false,
        })
    });
})