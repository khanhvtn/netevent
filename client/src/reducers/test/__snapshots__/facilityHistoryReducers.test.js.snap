// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FACILITY HISTORY REDUCER Should return FACILITY_HISTORY_LOADING 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "facilityHistories": Array [],
  "isLoading": undefined,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`FACILITY HISTORY REDUCER Should return default state 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "facilityHistories": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;
